Aquarius/Src/Aquarius/Sound/Devices.cpp

```diff
@@ -0,0 +1,81 @@
+#include "Devices.h"
+#include "Aquarius/Core/Log.h"
+#include <AL/alc.h>
+
+
+namespace Aquarius {
+
+    namespace Sound {
+
+        Device::Device()
+        {
+            Initialize();
+        }
+
+        void Device::Initialize()
+        {
+            m_Device = alcOpenDevice(nullptr);
+
+            if (!m_Device)
+            {
+                AQ_CORE_ERROR("Failed to open a sound device");
+                return;
+            }
+
+            else
+            {
+                AQ_CORE_INFO("A sound device was successfully opened");
+            }
+
+            m_Context = alcCreateContext(m_Device, nullptr);
+
+            if (!m_Context)
+            {
+                AQ_CORE_ERROR("Failed to create the sound context");
+            }
+
+            else
+                {
+                AQ_CORE_INFO("Sound context created successfully");
+            }
+
+            if (!alcMakeContextCurrent(m_Context))
+            {
+                AQ_CORE_ERROR("Failed to make the context current");
+            }
+        }
+
+        Device * Device::get()
+        {
+            static Device* device = new Device();
```

Comment:
I am not getting any more errors now when this object gets destroyed, but I don't think this will ever get deleted (there is no call to delete on this Device allocated on the heap even though it will only ever get allocated once). It might be better to store this static Device* in the class, and then in get you can check if it is nullptr, and if it is then you can create the Device, and then in the destructor you can delete it, if that makes sense